index.rst

```diff
@@ -58,24 +58,316 @@ We describe the design and implementation of the LSST Alert Distribution System,
+
+
 
 Alert Database
 ==============
 
 Deployment
 ===========
 
-.. Add content here.
-.. Do not include the document title (it's automatically added from metadata.yaml).
+The deployment of the alert distribution and mini-broker described here differs
+from the prototype described in `DMTN-081 <https://dmtn-081.lsst.io>`_
+and has been revised with significant improvement in design.
+DMTN-081 describes a system in which groups of filters are deployed
+in a single container, sharing one copy of the full alert stream
+and utilizing Python's multiprocessing module to parallelize the filters.
+Having a group of filters share a copy of the full alert stream
+minimizes the number of full streams and the amount of data
+that needs to be pulled from Kafka, which is necessary given
+a cap on the network bandwidth available for the system.
+Additionally, the load of a large number (~10) of consumers,
+i.e. a large number of full streams, may cause filters to lag
+and not be able to keep up with the volume and real-time velocity of alerts.
+(See DMTN-028.)
+
+The previous design deploys filters in groups in order to avoid the
+bandwidth needed to otherwise support the design of a one-stream-per-filter
+(and therefore 100+ streams) on a single Kafka instance.
+The major drawback in this deployment is that groups of filters are running
+within a single container and within a single executed Python script.
+Filters cannot be isolated to their own environments, and one filter
+may slow or otherwise affect another.
+
+The design described here attempts to avoid both the bandwidth issue
+and the issue of shared filter environments by placing
+components methodically on the hardware available to the system.
+This design separates each filter into its own Docker container so that
+filters should have no effect on each other.
+However, instead of having filter containers deployed on potentially
+many separate nodes all consuming streams from one Kafka instance running
+on its own dedicated node, a system of downstream Kafka mirrors are deployed
+on several nodes from which co-located filters can read streams locally,
+cutting down on the data throughput between nodes.
+Filters run in separate Docker containers which are placed on the
+same node as a Kafka instance mirroring the full stream pulled from the
+upstream central Kafka hub where alerts are sent.
+The number of mirrored Kafka instances / groups of filters able to
+be supported is then determined by the node-to-node available bandwidth,
+but the filters reading from the downstream local Kafka instances
+can take advantage of the internal bandwidth of the node on which they run.
+
+The updated deployment plan of the alert distribution system
+including filtering with the mini-broker is shown in the figure below.
+Content of the alerts is created in science pipelines, potentially
+parallelized by CCD.
+In the alert_stream repo, AlertProducers serialize Python dicts
```

Comment:
Are these `dict`s representing the outputs of the pipelines? Sorry if that should be obvious...! If so, note that our pipelines won't actually produce output as `dict`s, so it may be worth just mentioning that this is a stand-in/mock functionality.

Reply:
Yes and ok.